auth.go

```diff
@@ -27,9 +27,12 @@ type authBase struct {
 	apiSecret string
 }
 
-func (b authBase) withAuth(ctx context.Context, grant auth.VideoGrant) (context.Context, error) {
+func (b authBase) withAuth(ctx context.Context, grant auth.VideoGrant, sip *auth.SIPGrant) (context.Context, error) {
```

Comment:
I think it'd be better to have two separate functions for these things, since we very rarely use these together

Reply:
i.e. `withSIPAuth`

Reply:
Updated, kept a single method, but used interface type as argument. Looks a lot nicer. WDYT?

Should probably be moved to `protocol/auth` eventually.